pkg/ring/lifecycler.go

```diff
@@ -27,6 +27,12 @@ var (
 	errInvalidTokensGeneratorStrategy = errors.New("invalid token generator strategy")
 )
 
+type LifecyclerDelegate interface {
+	// OnRingInstanceHeartbeat is called while the instance is updating its heartbeat
+	// in the ring.
+	OnRingInstanceHeartbeat(lifecycler *Lifecycler, ctx context.Context)
```

Comment:
I'd love to see that the same function parameter to be reused. The current interface is a bit weird and we are not using the lifecycler at all in the compactor implementation